src/main/java/mops/businesslogic/DirectoryServiceImpl.java

```diff
@@ -0,0 +1,253 @@
+package mops.businesslogic;
+
+import lombok.AllArgsConstructor;
+import mops.businesslogic.exception.DatabaseException;
+import mops.businesslogic.exception.DeleteAccessPermissionException;
+import mops.exception.MopsException;
+import mops.persistence.DirectoryPermissionsRepository;
+import mops.persistence.DirectoryRepository;
+import mops.persistence.directory.Directory;
+import mops.persistence.file.FileInfo;
+import mops.persistence.permission.DirectoryPermissionEntry;
+import mops.persistence.permission.DirectoryPermissions;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+@SuppressWarnings("PMD.TooManyMethods") //this class needs more methods
+public class DirectoryServiceImpl implements DirectoryService {
+
+    /**
+     * Represents the role of an admin.
+     */
+    public static final String ADMINISTRATOR = "administrator";
+    /**
```

Comment:
```suggestion
    public static final String ADMINISTRATOR = "administrator";
    
    /**
```
Would add empty row here for formatting